R/utilities-simulation.R

```diff
@@ -43,3 +43,38 @@ runSimulation <- function(simulation) {
   results <- clrCall(simulationRunner, "RunSimulation", simulation$ref)
   return(clrCall(results, "IndividualResultsAsArray"))
 }
+
+
+#' Adds the quantities as output into the  \code{simulation}. The quantiites can either be specified using explicit instances or using paths.
+#'
+#' @param quantities Instance of quantities to add (typically retrieved using \code{getAllQuantitiesMatching})
+#' @param paths List of quantity path to add
+#' @param simulation Instance of a simulation for which output selection should be updated.
+#'
+#' @examples
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' outputs <- c("Organism|VenousBlood|Plasma|Caffeine", "Organism|ArterialBlood|**|Caffeine")
+#' addOutputs(paths=outputs, sim)
+#'
+#' @export
+addOutputs <- function(quantities = NULL, paths=NULL, simulation){
+  quantities <- c(quantities)
+  paths <- c(paths)
+
+  validateIsOfType(simulation, "Simulation")
+  validateIsOfType(quantities, "Quantity", nullAllowed = TRUE)
+  validateIsOfType(paths, "character", nullAllowed = TRUE)
+  validateNotAllNull(quantities, paths)
+
+  quantities <- uniqueEntities(c(quantities, getAllQuantitiesMatching(paths, simulation)), compareBy = "path")
+
+  outputSelections <- simulation$settings$outputSelections
+
+  for (quantity in quantities) {
+    outputSelections$addQuantity(quantity)
+  }
+
+  return(quantities)

```

Comment:
I think this method should not return anything, or silently the simulation.

Reply:
@PavelBal Why? If we do not return the selected quantities, there is no way for the user to actually verify what outputs were automatically added (when using wild cards selection for example)

If the actual instances are returned, you can actually print them. As a user of this method, you are free to forget about the returnes value obviously

Reply:
But I agree. The returned value should be silent!